File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/datasync/relationship/CreateRelationship.java
package com.pubnub.api.java.endpoints.datasync.relationship;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.datasync.relationship.PNDataSyncCreateRelationshipResult;
import org.jetbrains.annotations.Nullable;

import java.util.Map;

/**
 * @see com.pubnub.api.java.datasync.DataSync#createRelationship(String, String, String, int)
 */
public interface CreateRelationship extends Endpoint<PNDataSyncCreateRelationshipResult> {
    /**
     * Optional relationship identifier. When not set the server generates one.
     */
    CreateRelationship relationshipId(@Nullable String relationshipId);

    /**
     * Optional relationship status.
     */
    CreateRelationship status(@Nullable String status);

    /**
     * Optional arbitrary JSON object payload.
     */
    CreateRelationship payload(@Nullable Map<String, Object> payload);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/datasync/relationship/CreateRelationship.kt
package com.pubnub.api.endpoints.datasync.relationship

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.datasync.relationship.PNDataSyncCreateRelationshipResult

/**
 * @see [com.pubnub.api.datasync.DataSync.createRelationship]
 */
actual interface CreateRelationship : Endpoint<PNDataSyncCreateRelationshipResult>
